### PR TITLE
Emit events when calendar is opened and navigated

### PR DESCRIPTION
**Pull request by arrenleung**
# Alaska Airlines Pull Request

**Resolves:** #195

## Summary:

- Adding `auroDatePicker-calendarOpened` event:
  - Emitted from `auro-datepicker` and contains details for the current month and year that is visible
- Adding `auroCalendar-monthChanged` event:
  - Emitted from `auro-calendar` and contains details for the direction of navigation (forward or backward) and the month and year that is now visible (left-most month for ranged datepicker)
- Fixing capitalization for the validated event: `auroDatepicker-validated` -> `auroDatePicker-validated`
- Updating docs to reflect new events added

## Type of change:

- [X] New capability

## Checklist:

- [X] My update follows the CONTRIBUTING guidelines of this project
- [X] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Comment by blackfalcon** [comment]
@arrenleung please REBASE your feature branch. Please see this doc if support is needed. https://auro.alaskaair.com/contributing/upstream

File: src/auro-datepicker.js [hunk 1]
```diff
@@ -345,6 +347,41 @@ export class AuroDatePicker extends LitElement {
...
+   */
+  notifyCalendarOpened() {
+    this.dispatchEvent(new CustomEvent('auroDatePicker-calendarOpened', {
+      bubbles: true,
+      composed: true,
+      detail: {
+        month: this.calendar.month,
+        year: this.calendar.year,
+        numCalendars: this.calendar.numCalendars,
+      },
```
Comment (jason-capsule42) [inline src/auro-datepicker.js:363]: Let's follow the same naming pattern as dropdown.

```suggestion
    this.dispatchEvent(new CustomEvent('auroDatePicker-toggled', {
      bubbles: true,
      composed: true,
      detail: {
        exanded: this.dropdown.isPopoverVisible
      },
```

I don't think we should include the other details in this event.

For you to extract the month, year and, numCalendars you have to query `event.detail.month` etc. anyhow. It's just as easy to do `datepicker.calendar.month` etc.

File: src/auro-calendar.js [hunk 2]
```diff
@@ -198,6 +199,7 @@ export class AuroCalendar extends RangeDatepicker {
     ) {
       if (changedProperties.has("month") || changedProperties.has("year")) {
         this.monthChanged(this.month, this.year);
+        this.notifyMonthChanged(this.month, this.year, this.numCalendars);
```
Comment (jason-capsule42) [inline src/auro-calendar.js:202]: Probably best to call `this.notifyMonthChanged()` from `this.monthChanged()`.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: src/auro-calendar.js (hunk 2 above)
Comment (arrenleung) [reply src/auro-calendar.js:202]: From my understanding, `monthChanged()` seems to be in the `RangeDatepicker` class from range-datepicker.js, and `this.numCalendars` is a property created in auro-calendar only. If we move it to  `monthChanged()` in range-datepicker.js, we wouldn't be able to pass the number of calendars in the event unless we change the parameters of the method to pass `this.numCalendars`. Let me know if you'd like to go down that route or keep it in auro-calendar.

File: src/auro-datepicker.js [hunk 3]
```diff
@@ -361,6 +396,9 @@ export class AuroDatePicker extends LitElement {
 
     this.dropdown.addEventListener('auroDropdown-toggled', () => {
       this.setAttribute('aria-expanded', this.dropdown.isPopoverVisible);
+      if (this.dropdown.isPopoverVisible) {
+        this.notifyDatepickerToggled();
+      }
```
Comment (jason-capsule42) [inline src/auro-datepicker.js:401]: We should fire this event when the bib is opened and when it's closed. The event includes the detail about `expanded`.

I would just remove the if statement wrapping this.

File: src/auro-calendar.js (hunk 2 above)
Comment (jason-capsule42) [reply src/auro-calendar.js:202]: Eventually we want to re-write the 3rd party tool anyhow here. Just copy the entire function as is from `RangeDatepicker` and then modify it to include the call to `notifyMonthChanged()`.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

**Pull request merged**